### ISSUE TITLE
Jupyter lab failed to build

### ISSUE THREAD

**Issue by kevinalexandr19** [jupyterlab/jupyterlab#9533]
## Description

Jupyter lab failed to build

## Reproduce

When I open jupyter-lab, it asks for build
Typing "jupyter lab build" in terminal fails and return this:

[LabBuildApp] JupyterLab 3.0.0
[LabBuildApp] Building in c:\python\share\jupyter\lab
[LabBuildApp] Building jupyterlab assets (production, minimized)
Build failed.
Troubleshooting: If the build failed due to an out-of-memory error, you
may be able to fix it by disabling the `dev_build` and/or `minimize` options.

If you are building via the `jupyter lab build` command, you can disable
these options like so:

jupyter lab build --dev-build=False --minimize=False

You can also disable these options for all JupyterLab builds by adding these
lines to a Jupyter config file named `jupyter_config.py`:

c.LabBuildApp.minimize = False
c.LabBuildApp.dev_build = False

If you don't already have a `jupyter_config.py` file, you can create one by
adding a blank file of that name to any of the Jupyter config directories.
The config directories can be listed by running:

jupyter --paths

Explanation:

- `dev-build`: This option controls whether a `dev` or a more streamlined
`production` build is used. This option will default to `False` (i.e., the
`production` build) for most users. However, if you have any labextensions
installed from local files, this option will instead default to `True`.
Explicitly setting `dev-build` to `False` will ensure that the `production`
build is used in all circumstances.

- `minimize`: This option controls whether your JS bundle is minified
during the Webpack build, which helps to improve JupyterLab's overall
performance. However, the minifier plugin used by Webpack is very memory
intensive, so turning it off may help the build finish successfully in
low-memory environments.

An error occured.
RuntimeError: JupyterLab failed to build
See the log file for details:  C:\Users\Kevin\AppData\Local\Temp\jupyterlab-debug-gco5jjey.log

### Content of log file:

[LabBuildApp] Building in c:\python\share\jupyter\lab
[LabBuildApp] Node v15.4.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node c:\python\lib\site-packages\jupyterlab\staging\yarn.js install --non-interactive
[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@2.1.3: The platform "win32" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning "@jupyterlab/extensionmanager > react-paginate@6.5.0" has incorrect peer dependency "react@^16.0.0".
warning "@jupyterlab/json-extension > react-highlighter@0.4.3" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/json-extension > react-json-tree@0.12.1" has incorrect peer dependency "react@^16.3.0".
warning "@jupyterlab/json-extension > react-json-tree@0.12.1" has incorrect peer dependency "react-dom@^16.3.0".
warning "@jupyterlab/ui-components > @blueprintjs/core > react-popper@1.3.7" has incorrect peer dependency "react@0.14.x || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/ui-components > @blueprintjs/core > react-popper > create-react-context@0.3.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/vdom > @nteract/transform-vdom@4.0.16-alpha.0" has incorrect peer dependency "react@^16.3.2".
warning " > @lumino/coreutils@1.5.3" has unmet peer dependency "crypto@1.0.1".
[5/5] Building fresh packages...
success Saved lockfile.
Done in 17.08s.

[LabBuildApp] > node c:\python\lib\site-packages\jupyterlab\staging\yarn.js yarn-deduplicate -s fewer --fail
[LabBuildApp] yarn run v1.21.1
$ c:\python\share\jupyter\lab\staging\node_modules\.bin\yarn-deduplicate -s fewer --fail
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] > node c:\python\lib\site-packages\jupyterlab\staging\yarn.js
[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@2.1.3: The platform "win32" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning "@jupyterlab/extensionmanager > react-paginate@6.5.0" has incorrect peer dependency "react@^16.0.0".
warning "@jupyterlab/json-extension > react-highlighter@0.4.3" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/json-extension > react-json-tree@0.12.1" has incorrect peer dependency "react@^16.3.0".
warning "@jupyterlab/json-extension > react-json-tree@0.12.1" has incorrect peer dependency "react-dom@^16.3.0".
warning "@jupyterlab/ui-components > @blueprintjs/core > react-popper@1.3.7" has incorrect peer dependency "react@0.14.x || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/ui-components > @blueprintjs/core > react-popper > create-react-context@0.3.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/vdom > @nteract/transform-vdom@4.0.16-alpha.0" has incorrect peer dependency "react@^16.3.2".
warning " > @lumino/coreutils@1.5.3" has unmet peer dependency "crypto@1.0.1".
[5/5] Building fresh packages...
Done in 4.08s.

[LabBuildApp] > node c:\python\lib\site-packages\jupyterlab\staging\yarn.js run build:prod:minimize
[LabBuildApp] yarn run v1.21.1
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] [31mTypeError: The 'compilation' argument must be an instance of Compilation
    at Function.getCompilationHooks (c:\python\share\jupyter\lab\staging\node_modules\webpack\lib\javascript\JavascriptModulesPlugin.js:119:10)
    at c:\python\share\jupyter\lab\staging\node_modules\terser-webpack-plugin\dist\index.js:566:67
    at _next35 (eval at create (c:\Python\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:40:1)
    at _next13 (eval at create (c:\Python\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:97:1)
    at Hook.eval [as call] (eval at create (c:\Python\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:127:1)
    at Hook.CALL_DELEGATE [as _call] (c:\Python\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:14:14)
    at Compiler.newCompilation (c:\Python\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:988:26)
    at c:\Python\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:1029:29
    at Hook.eval [as callAsync] (eval at create (c:\Python\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (c:\Python\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:18:14)[39m
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "c:\python\lib\site-packages\jupyterlab\debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "c:\python\lib\site-packages\jupyterlab\labapp.py", line 168, in start
    raise e

[LabBuildApp]   File "c:\python\lib\site-packages\jupyterlab\labapp.py", line 164, in start
    build(name=self.name, version=self.version,

[LabBuildApp]   File "c:\python\lib\site-packages\jupyterlab\commands.py", line 463, in build
    return handler.build(name=name, version=version, static_url=static_url,

[LabBuildApp]   File "c:\python\lib\site-packages\jupyterlab\commands.py", line 675, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

[LabBuildApp] Exiting application: JupyterLab

## Context

- Operating System and version: Windows 10 Pro
- Browser and version: Google Chrome v 87.0.4280.88
- JupyterLab version: 3.0.0

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>
<pre>
Paste the output from your browser Javascript console here.
</pre>
</details>

**Comment by kevinalexandr19** [comment]
I also want to state that I installed jupyterlab for the 1st time on Dec 21st (can't remember the exact version I think it was 3.0.0 as well).
The problem appeared 2 days ago when I tried to install a new extension; jupyter lab build failed and then I uninstalled/reinstalled jupyterlab but still got the error from above.

**Comment by serial-lens** [comment]
Seeing the exact same issue. Windows 10, Google Chrome, JupyterLab 3.0.0

```
Log file:
[LabBuildApp] Building in c:\python_environments\opencv\share\jupyter\lab
[LabBuildApp] Node v15.5.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node c:\python_environments\opencv\lib\site-packages\jupyterlab\staging\yarn.js install --non-interactive
[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.42s.

[LabBuildApp] > node c:\python_environments\opencv\lib\site-packages\jupyterlab\staging\yarn.js yarn-deduplicate -s fewer --fail
[LabBuildApp] yarn run v1.21.1
$ c:\python_environments\opencv\share\jupyter\lab\staging\node_modules\.bin\yarn-deduplicate -s fewer --fail
Done in 0.28s.

[LabBuildApp] > node c:\python_environments\opencv\lib\site-packages\jupyterlab\staging\yarn.js run build:prod:minimize
[LabBuildApp] yarn run v1.21.1
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] [31mTypeError: The 'compilation' argument must be an instance of Compilation
    at Function.getCompilationHooks (c:\python_environments\opencv\share\jupyter\lab\staging\node_modules\webpack\lib\javascript\JavascriptModulesPlugin.js:119:10)
    at c:\python_environments\opencv\share\jupyter\lab\staging\node_modules\terser-webpack-plugin\dist\index.js:566:67
    at _next35 (eval at create (c:\Python_Environments\opencv\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:40:1)
    at _next13 (eval at create (c:\Python_Environments\opencv\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:97:1)
    at Hook.eval [as call] (eval at create (c:\Python_Environments\opencv\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:127:1)
    at Hook.CALL_DELEGATE [as _call] (c:\Python_Environments\opencv\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:14:14)
    at Compiler.newCompilation (c:\Python_Environments\opencv\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:988:26)
    at c:\Python_Environments\opencv\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:1029:29
    at Hook.eval [as callAsync] (eval at create (c:\Python_Environments\opencv\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (c:\Python_Environments\opencv\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:18:14)[39m
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "c:\python_environments\opencv\lib\site-packages\jupyterlab\debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "c:\python_environments\opencv\lib\site-packages\jupyterlab\labapp.py", line 168, in start
    raise e

[LabBuildApp]   File "c:\python_environments\opencv\lib\site-packages\jupyterlab\labapp.py", line 164, in start
    build(name=self.name, version=self.version,

[LabBuildApp]   File "c:\python_environments\opencv\lib\site-packages\jupyterlab\commands.py", line 463, in build
    return handler.build(name=name, version=version, static_url=static_url,

[LabBuildApp]   File "c:\python_environments\opencv\lib\site-packages\jupyterlab\commands.py", line 675, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

[LabBuildApp] Exiting application: JupyterLab
```

**Comment by myWillison** [comment]
Facing the same issue, too. I'm trying to install jupyterlab extensions. Windows 10, Python 3.9.1, Jupyter 3.0.0, Node 15.5.0
```
[LabBuildApp] Building in c:\users\willi\appdata\local\programs\python\python39\share\jupyter\lab
[LabBuildApp] Node v15.5.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node c:\users\willi\appdata\local\programs\python\python39\lib\site-packages\jupyterlab\staging\yarn.js install --non-interactive
[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.52s.

[LabBuildApp] > node c:\users\willi\appdata\local\programs\python\python39\lib\site-packages\jupyterlab\staging\yarn.js yarn-deduplicate -s fewer --fail
[LabBuildApp] yarn run v1.21.1
$ c:\users\willi\appdata\local\programs\python\python39\share\jupyter\lab\staging\node_modules\.bin\yarn-deduplicate -s fewer --fail
Done in 0.41s.

[LabBuildApp] > node c:\users\willi\appdata\local\programs\python\python39\lib\site-packages\jupyterlab\staging\yarn.js run build:prod:minimize
[LabBuildApp] yarn run v1.21.1
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] [31mTypeError: The 'compilation' argument must be an instance of Compilation
    at Function.getCompilationHooks (c:\users\willi\appdata\local\programs\python\python39\share\jupyter\lab\staging\node_modules\webpack\lib\javascript\JavascriptModulesPlugin.js:119:10)
    at c:\users\willi\appdata\local\programs\python\python39\share\jupyter\lab\staging\node_modules\terser-webpack-plugin\dist\index.js:566:67
    at _next35 (eval at create (c:\Users\willi\AppData\Local\Programs\Python\Python39\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:40:1)
    at _next13 (eval at create (c:\Users\willi\AppData\Local\Programs\Python\Python39\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:97:1)
    at Hook.eval [as call] (eval at create (c:\Users\willi\AppData\Local\Programs\Python\Python39\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:127:1)
    at Hook.CALL_DELEGATE [as _call] (c:\Users\willi\AppData\Local\Programs\Python\Python39\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:14:14)
    at Compiler.newCompilation (c:\Users\willi\AppData\Local\Programs\Python\Python39\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:988:26)
    at c:\Users\willi\AppData\Local\Programs\Python\Python39\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:1029:29
    at Hook.eval [as callAsync] (eval at create (c:\Users\willi\AppData\Local\Programs\Python\Python39\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (c:\Users\willi\AppData\Local\Programs\Python\Python39\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:18:14)[39m
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "c:\users\willi\appdata\local\programs\python\python39\lib\site-packages\jupyterlab\debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "c:\users\willi\appdata\local\programs\python\python39\lib\site-packages\jupyterlab\labapp.py", line 168, in start
    raise e

[LabBuildApp]   File "c:\users\willi\appdata\local\programs\python\python39\lib\site-packages\jupyterlab\labapp.py", line 164, in start
    build(name=self.name, version=self.version,

[LabBuildApp]   File "c:\users\willi\appdata\local\programs\python\python39\lib\site-packages\jupyterlab\commands.py", line 463, in build
    return handler.build(name=name, version=version, static_url=static_url,

[LabBuildApp]   File "c:\users\willi\appdata\local\programs\python\python39\lib\site-packages\jupyterlab\commands.py", line 675, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

[LabBuildApp] Exiting application: JupyterLab
```

**Comment by snoehmer** [comment]
Same issue here, also while rebuilding after installing extensions. Windows 10, Python 3.9.1, Jupyter Lab 3.0.1, Node v15.
Also tried on a different machine (Windows 10, Python 3.9.1, Jupyter Lab 3.0.1, Node v12) in a Python virtual environment with the same result:
```
[LabBuildApp] Building in c:\users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab
[LabBuildApp] Node v12.19.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\staging\yarn.js install --non-interactive
[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@2.1.3: The platform "win32" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning "@jupyterlab/extensionmanager > react-paginate@6.5.0" has incorrect peer dependency "react@^16.0.0".
warning "@jupyterlab/json-extension > react-highlighter@0.4.3" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/json-extension > react-json-tree@0.12.1" has incorrect peer dependency "react@^16.3.0".
warning "@jupyterlab/json-extension > react-json-tree@0.12.1" has incorrect peer dependency "react-dom@^16.3.0".
warning "@jupyterlab/ui-components > @blueprintjs/core > react-popper@1.3.7" has incorrect peer dependency "react@0.14.x || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/ui-components > @blueprintjs/core > react-popper > create-react-context@0.3.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/vdom > @nteract/transform-vdom@4.0.16-alpha.0" has incorrect peer dependency "react@^16.3.2".
warning " > @lumino/coreutils@1.5.3" has unmet peer dependency "crypto@1.0.1".
[5/5] Building fresh packages...
Done in 88.86s.

[LabBuildApp] > node c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\staging\yarn.js yarn-deduplicate -s fewer --fail
[LabBuildApp] yarn run v1.21.1
$ c:\users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\.bin\yarn-deduplicate -s fewer --fail
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] > node c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\staging\yarn.js
[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@2.1.3: The platform "win32" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning "@jupyterlab/extensionmanager > react-paginate@6.5.0" has incorrect peer dependency "react@^16.0.0".
warning "@jupyterlab/json-extension > react-highlighter@0.4.3" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/json-extension > react-json-tree@0.12.1" has incorrect peer dependency "react@^16.3.0".
warning "@jupyterlab/json-extension > react-json-tree@0.12.1" has incorrect peer dependency "react-dom@^16.3.0".
warning "@jupyterlab/ui-components > @blueprintjs/core > react-popper@1.3.7" has incorrect peer dependency "react@0.14.x || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/ui-components > @blueprintjs/core > react-popper > create-react-context@0.3.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@jupyterlab/vdom > @nteract/transform-vdom@4.0.16-alpha.0" has incorrect peer dependency "react@^16.3.2".
warning " > @lumino/coreutils@1.5.3" has unmet peer dependency "crypto@1.0.1".
[5/5] Building fresh packages...
Done in 12.16s.

[LabBuildApp] > node c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\staging\yarn.js run build:prod:minimize
[LabBuildApp] yarn run v1.21.1
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] [31mTypeError: The 'compilation' argument must be an instance of Compilation
    at Function.getCompilationHooks (c:\users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\webpack\lib\javascript\JavascriptModulesPlugin.js:119:10)
    at c:\users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\terser-webpack-plugin\dist\index.js:566:67
    at _next35 (eval at create (c:\Users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:38:1)
    at _next13 (eval at create (c:\Users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:95:1)
    at Hook.eval [as call] (eval at create (c:\Users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:125:1)
    at Hook.CALL_DELEGATE [as _call] (c:\Users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:14:14)
    at Compiler.newCompilation (c:\Users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:988:26)
    at c:\Users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:1029:29
    at Hook.eval [as callAsync] (eval at create (c:\Users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:4:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (c:\Users\nos\python-virtual-environments\env_jupyter\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:18:14)[39m
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\labapp.py", line 168, in start
    raise e

[LabBuildApp]   File "c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\labapp.py", line 164, in start
    build(name=self.name, version=self.version,

[LabBuildApp]   File "c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\commands.py", line 463, in build
    return handler.build(name=name, version=version, static_url=static_url,

[LabBuildApp]   File "c:\users\nos\python-virtual-environments\env_jupyter\lib\site-packages\jupyterlab\commands.py", line 675, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

[LabBuildApp] Exiting application: JupyterLab

```

**Comment by daizhirui** [comment]
Same issue with JupyterLab 3.0.3 and Node v15.5.1 on Windows 10

```
[LabBuildApp] Building in c:\program files\python38\share\jupyter\lab
[LabBuildApp] Node v15.5.1

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node "c:\program files\python38\lib\site-packages\jupyterlab\staging\yarn.js" install --non-interactive
[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.37s.

[LabBuildApp] > node "c:\program files\python38\lib\site-packages\jupyterlab\staging\yarn.js" yarn-deduplicate -s fewer --fail
[LabBuildApp] yarn run v1.21.1
$ "c:\program files\python38\share\jupyter\lab\staging\node_modules\.bin\yarn-deduplicate" -s fewer --fail
Done in 0.24s.

[LabBuildApp] > node "c:\program files\python38\lib\site-packages\jupyterlab\staging\yarn.js" run build:prod:minimize
[LabBuildApp] yarn run v1.21.1
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] [31mTypeError: The 'compilation' argument must be an instance of Compilation
    at Function.getCompilationHooks (c:\program files\python38\share\jupyter\lab\staging\node_modules\webpack\lib\javascript\JavascriptModulesPlugin.js:119:10)
    at c:\program files\python38\share\jupyter\lab\staging\node_modules\terser-webpack-plugin\dist\index.js:566:67
    at _next35 (eval at create (c:\Program Files\Python38\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:40:1)
    at _next13 (eval at create (c:\Program Files\Python38\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:97:1)
    at Hook.eval [as call] (eval at create (c:\Program Files\Python38\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:127:1)
    at Hook.CALL_DELEGATE [as _call] (c:\Program Files\Python38\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:14:14)
    at Compiler.newCompilation (c:\Program Files\Python38\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:988:26)
    at c:\Program Files\Python38\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:1030:29
    at Hook.eval [as callAsync] (eval at create (c:\Program Files\Python38\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (c:\Program Files\Python38\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:18:14)[39m
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "c:\program files\python38\lib\site-packages\jupyterlab\debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "c:\program files\python38\lib\site-packages\jupyterlab\labapp.py", line 168, in start
    raise e

[LabBuildApp]   File "c:\program files\python38\lib\site-packages\jupyterlab\labapp.py", line 164, in start
    build(name=self.name, version=self.version,

[LabBuildApp]   File "c:\program files\python38\lib\site-packages\jupyterlab\commands.py", line 463, in build
    return handler.build(name=name, version=version, static_url=static_url,

[LabBuildApp]   File "c:\program files\python38\lib\site-packages\jupyterlab\commands.py", line 675, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

[LabBuildApp] Exiting application: JupyterLab
```

**Comment by jasongrout** [comment]
> [webpack-cli] �[31mTypeError: The 'compilation' argument must be an instance of Compilation

That error makes me think that doing `jupyter lab clean` may solve the issue for you. Can you try doing `jupyter lab clean`?

The error indicates to me that an upgrade overwrote some files from jlab 2, but not all, and the compilation is getting mixed up having both jlab 2 and jlab 3 files around.

Technical note: On an upgrade, we should probably put a sentinel file in the build directory that indicates an upgrade happened and we should do a `jupyter lab clean` automatically on the next jupyter lab start up. The clean would then remove this sentinel file.

**Comment by daizhirui** [comment]
> > [webpack-cli] �[31mTypeError: The 'compilation' argument must be an instance of Compilation
> 
> That error makes me think that doing `jupyter lab clean` may solve the issue for you. Can you try doing `jupyter lab clean`?
> 
> The error indicates to me that an upgrade overwrote some files from jlab 2, but not all, and the compilation is getting mixed up having both jlab 2 and jlab 3 files around.
> 
> Technical note: On an upgrade, we should probably put a sentinel file in the build directory that indicates an upgrade happened and we should do a `jupyter lab clean` automatically on the next jupyter lab start up. The clean would then remove this sentinel file.

I tried `jupyter lab clean`. I am sorry that it doesn't help. I still get the same error.

**Comment by daizhirui** [comment]
I find a trick to solve this problem: run `npm install` in the `<conda_folder>\share\jupyter\lab\staging` folder. Then, run `jupyter lab build` again.

**Comment by potoo0** [comment]
> I find a trick to solve this problem: run `npm install` in the `<conda_folder>\share\jupyter\lab\staging` folder. Then, run `jupyter lab build` again.

thx, solved.
my env:
- conda: 4.9.2
- python: 3.8.5
- jupyterlab: 3.0.5
- jupyterlab plugins:
  - jupyter-matplotlib v0.8.2 enabled ok
  - jupyterlab-plotly v4.14.3 enabled ok
  - @ryantam626/jupyterlab_code_formatter v1.4.2 enabled ok

**Comment by daubweb** [comment]
same issue here, tried the npm install trick, did not work. I am using jupyterlab-plotly as well... might this be the culprit?

my env:
* pipenv / virtualenv
* python: 3.9.1
* jupyterlab: 3.0.5
* Windows 10 Pro

**Comment by gc202** [comment]
@daubweb I'm also thinking that it's the new jupyterlab-plotly release. I'm having the same issue since trying to update it.

**Comment by daubweb** [comment]
I think that it's somehow related to webpack 5 and html-webpack-plugin, as
this issue now also occurs on others packages. Tried to find a fix all
day...

gc202 <notifications@github.com> schrieb am Mo., 18. Jan. 2021, 17:52:

**Comment by zagi** [comment]
This issue is making me a headache for several past days 🤯. The mentioned `npm install` trick didn't work. Has anyone overcome it?

**Comment by stromnov** [comment]
> This issue is making me a headache for several past days 🤯. The mentioned `npm install` trick didn't work. Has anyone overcome it?

works for me: https://stackoverflow.com/a/64580815

**Comment by zagi** [comment]
> > This issue is making me a headache for several past days 🤯. The mentioned `npm install` trick didn't work. Has anyone overcome it?
> 
> works for me: https://stackoverflow.com/a/64580815

thx. unfortunately, it doesn't work for me.

**Comment by jasongrout** [comment]
@zagi, can you post a reproducible set of steps from a clean environment giving the error you are seeing?

**Comment by jasongrout** [comment]
@SachinCompton made some excellent progress towards identifying what can cause the errors people are seeing here. Since we've made a lot of progress on that issue towards a solution, let's close this as a duplicate and move discussion over there.

For those that are wondering, the short version of our conclusions there is that it seems that sometimes there are two copies of webpack in the build, probably from an older extension or an extension that needs to be updated or something (still trying to figure that out). It may be that we can upgrade one of our packages to alleviate the symptoms - looking into that. For the time being, it would be helpful if those that are consistently seeing the issue can uninstall all of their extensions, then install them back one by one to try to get a reproducible test example.

**Comment by jasongrout** [comment]
The fix in #9651 was released in jlab 3.0.6

**Comment by richdevboston** [comment]
> 
> 
> @daubweb I'm also thinking that it's the new jupyterlab-plotly release. I'm having the same issue since trying to update it.
Same issue with jupyterlab-plotly !!    I have had dozens of problems with plotly in Jupyter Classic and JupyterLab.  They don't maintain their open source kits or demonstrations.   Plotly gets 3 thumbs down.
